fili-core/src/main/java/com/yahoo/bard/webservice/druid/client/impl/AsyncDruidWebServiceImpl.java

```diff
@@ -267,10 +270,11 @@ public void onThrowable(Throwable t) {
             LOG.error("druid {} http request failed: ", serviceConfig.getNameAndUrl(), t);
             failure.invoke(t);
         }
+        return new FailedFuture<>(new RuntimeException("Failed to reach druid"));
```

Comment:
This should happen inside the `catch` block, and the exception it's constructed with should make sure it wraps the exception that the `catch` is catching. (It's OK to wrap that exception and give it an extra message if you wish, but the causal chain needs to be preserved.